vita3k/modules/SceJpeg/SceJpegUser.cpp

```diff
@@ -131,41 +181,46 @@ EXPORT(int, sceJpegDecodeMJpeg, const unsigned char *pJpeg, SceSize isize, uint8
     int decodeMode, void *pTempBuffer, SceSize tempBufferSize, void *pCoefBuffer, SceSize coefBufferSize) {
     TRACY_FUNC(sceJpegDecodeMJpeg, pJpeg, isize, pRGBA, osize, decodeMode, pTempBuffer, tempBufferSize, pCoefBuffer, coefBufferSize);
 
-    if (decodeMode & SCE_JPEG_MJPEG_DOWNSCALE_ANY)
-        return STUBBED("JPEG downscaling is not implemented");
-
     const auto state = emuenv.kernel.obj_store.get<MJpegState>();
-
-    DecoderSize size = {};
+    configureDecoder(state, decodeMode);
 
     // the yuv data will always be smaller than the rgba data, so osize is an upper bound
     std::vector<uint8_t> temporary(osize);
+    DecoderSize size = {};
 
     state->decoder->send(pJpeg, isize);
     state->decoder->receive(temporary.data(), &size);
 
-    convert_yuv_to_rgb(temporary.data(), pRGBA, size.pitch_width, size.pitch_height, state->decoder->get_color_space());
+    SceJpegPitch yuv_pitch[4];
+    state->decoder->get_pitch_info(yuv_pitch);
+
+    if (isUnsupportedImageSize(yuv_pitch[0].x, yuv_pitch[0].y)) {
+        return RET_ERROR(SCE_JPEG_ERROR_UNSUPPORT_IMAGE_SIZE);
+    }
+
+    convert_yuv_to_rgb(temporary.data(), pRGBA, size.width, size.height, state->decoder->get_color_space(), false, yuv_pitch);
```

Comment:
size contains the pre-downscaled size, while I believe we want the post-downscaled dimension here, right?

Reply:
That's correct. I missed it because there were no issues during testing, but the post-downscaled size should be used there.
It seems that due to internal implementation characteristics of FFmpeg, it coincidentally worked without problems. (It probably allocated a larger workspace and then processed only the space actually occupied by the image.) It's right to correct this.